### ISSUE TITLE
Redis doesn't seem to function correctly

### ISSUE THREAD

**Issue by webpigeon** [spantaleev/matrix-docker-ansible-deploy#1140]
Hi,

I've been investigating issues with federation, and I think there is a bug in the redis image. The logs for the server show the following error from redis:

```
Jun 23 21:35:33 matrix matrix-redis[11142]: 8822:C 23 Jun 2021 21:35:33.068 # Failed opening the RDB file dump.rdb (in server root dir /data) for saving: Permission denied
```

This is causing the workers to go bananas. As far as I can tell, the /data directory in the container is owned by redis (on my system user id 999). Redis is running as the matrix user, which looks to be what's expected.

If I inspect the docker container, the /data/ directory seems to just be a random volume:

docker inspect matrix-redis

```
...
        "Mounts": [
            {
                "Type": "bind",
                "Source": "/matrix/redis/redis.conf",
                "Destination": "/usr/local/etc/redis/redis.conf",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "volume",
                "Name": "6343b58b7b4c5fa189a0bcf6eb12ba33148c5378bad90efcb4c30fbd3f2025f0",
                "Source": "/var/lib/docker/volumes/6343b58b7b4c5fa189a0bcf6eb12ba33148c5378bad90efcb4c30fbd3f2025f0/_data",
                "Destination": "/data",
                "Driver": "local",
                "Mode": "",
                "RW": true,
                "Propagation": ""
            }
        ],
...
```

There is a redis data folder which gets created, is seems to be designed to be managed by the role, but as far as I can tell it's never actually mapped when the service is started:

https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/1fb54a37cba126235b1688bb5e96226a65b4fe80/roles/matrix-redis/templates/systemd/matrix-redis.service.j2#L22

Because the folder is owned by the wrong user, the redis server therefore can't actually save anything and therefore refuses to do anything useful on disk (which looks to be causing my federation issues).

My question is:

Is this a bug (and it should be mapped) or have I misunderstood what's going on here?

**Comment by webpigeon** [comment]
I added the mapping using `matrix_redis_container_extra_arguments` to test it, looks like it is a bug because fixing it caused federation to start working properly.

**Comment by spantaleev** [comment]
I wonder why we weren't mounting anything to `/data` until now and why it wasn't such a big problem until now.

Fixed by 2d649db.

Thank you! :+1:
